lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -377,7 +377,7 @@ namespace utf8
             return ptr;
     }
 
-    void DecodeInto(__out_ecount_full(cch) char16 *buffer, LPCUTF8 ptr, size_t cch, DecodeOptions options)
+    void DecodeInto(__out_ecount_full(cch) char16 *buffer, LPCUTF8 ptr, LPCUTF8 end, size_t cch, DecodeOptions options)
```

Comment:
Maybe not worth maintaining 2 sets of functions. One set can be easily implemented by calling the other.